test/unit/test_matrix.cpp

```diff
@@ -0,0 +1,365 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include "dlaf/matrix.h"
+
+#include "gtest/gtest.h"
+#include "dlaf_test/util_matrix.h"
+#include "dlaf_test/util_types.h"
+
+using namespace dlaf;
+using namespace dlaf_test;
+using namespace matrix_test;
+using namespace testing;
+
+template <typename Type>
+class MatrixTest : public ::testing::Test {};
+
+TYPED_TEST_CASE(MatrixTest, MatrixElementTypes);
+
+std::vector<GlobalElementSize> sizes({{31, 17}, {29, 41}, {0, 1}, {3, 0}});
+std::vector<TileElementSize> block_sizes({{7, 11}, {13, 11}, {3, 3}});
+
+TYPED_TEST(MatrixTest, Constructor) {
+  using Type = TypeParam;
+  auto el = [](const GlobalElementIndex& index) {
+    SizeType i = index.row();
+    SizeType j = index.col();
+    return TypeUtilities<Type>::element(i + 0.001 * j, j - 0.01 * i);
+  };
+
+  for (const auto& size : sizes) {
+    for (const auto& block_size : block_sizes) {
+      Matrix<Type, Device::CPU> mat(size, block_size);
+
+      EXPECT_EQ(MatrixBase(size, block_size), mat);
+
+      set(mat, el);
+
+      CHECK_MATRIX_EQ(el, mat);
+    }
+  }
+}
+
+template <class T, Device device>
+void checkFromExisting(T* p, const matrix::LayoutInfo& layout, Matrix<T, device>& matrix) {
+  auto el = [](const GlobalElementIndex& index) {
```

Comment:
Please, use better names for `el` and `el2`, e.g. `reference_matrix1`, `reference_matrix2`.

Reply:
`reference_matrix` - > I would expect a matrix.
This is how `el` was chosen:
`matrix_elements` ~> `elements` ~>`el`
Is `elements` ok?

Reply:
It was just my opinion. I agree with you that `elements` does not change so much from `el`.